hs-bindgen/src/HsBindgen/Backend/TH.hs

```diff
@@ -66,11 +101,23 @@ instance TH.Quote q => BackendRep (BE q) where
       resolveP HsPrimCDouble = ''Foreign.C.Types.CDouble
       resolveP HsPrimCBool   = ''Foreign.C.Types.CBool
 
+  resolveLocal _ nm = TH.mkName (Text.unpack $ getHsName nm)
   mkExpr be = \case
       EGlobal n     -> TH.varE (resolve be n)
       EVar x        -> TH.varE (getFresh x)
       ECon n        -> hsConE n
-      EInt i        -> TH.litE (TH.IntegerL $ fromIntegral i)
+      EIntegral i _ -> TH.litE (TH.IntegerL i)
+      -- TH doesn't have floating-point literals, because it represents them
+      -- using the Rational type, which is incorrect.
```

Comment:
is there a GHC issue about this?

Reply:
I found [#13124](https://gitlab.haskell.org/ghc/ghc/-/issues/13124). I added a reference.